MdeModulePkg/Core/PiSmmCore/PiSmmIpl.c

```diff
@@ -752,11 +739,98 @@ MmCommunicationMmCommunicate3 (
   IN OUT VOID                              *CommBufferVirtual
   )
 {
-  return SmmCommunicationCommunicate (
-           &mSmmCommunication,
-           CommBufferPhysical,
-           NULL
-           );
+  EFI_STATUS                    Status;
+  EFI_MM_COMMUNICATE_HEADER_V3  *CommunicateHeader;
+  BOOLEAN                       OldInSmm;
+  UINT64                        MinCommSize;
+  UINT64                        TempCommSize;
+
+  //
+  // Check parameters
+  //
+  if ((CommBufferPhysical == NULL) || (CommBufferVirtual == NULL)) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  CommunicateHeader = (EFI_MM_COMMUNICATE_HEADER_V3 *)CommBufferVirtual;

```

Comment:
is `CommunicateHeader->HeaderGuid` supposed to be `gEfiMmCommunicateHeaderV3Guid` in the input buffer? Should that be validated here before consuming data from the structure?

Reply:
I agree it's not critical either so please feel free to wait. It's one of those inputs that just seem simple to validate before kicking off the full MM invocation process.

Reply:
Okay, will add it in the next wave of the edk2 PR. Thanks for the suggestion.